source/server/filter_chain_manager_impl.h

```diff
@@ -36,9 +88,26 @@ class FilterChainManagerImpl : public Network::FilterChainManager,
 
   void addFilterChain(
       absl::Span<const envoy::config::listener::v3alpha::FilterChain* const> filter_chain_span,
-      FilterChainFactoryBuilder& b);
+      FilterChainFactoryBuilder& b, FilterChainFactoryContextCallback& callback);
   static bool isWildcardServerName(const std::string& name);
 
+  std::unique_ptr<FilterChainFactoryContextCallback>
+  createFilterChainFactoryContextCallback(Configuration::FactoryContext& parent_context);
+
+  // A naive implementation which commits the context as soon as the context is created.
```

Comment:
What does this mean? Why do we need this? If possible can you just remove this from this PR?

Reply:
This callback provides compact view of FilterChainManagerImpl which manages the lifetime of filter chain context. In the next PR FilterChainManagerImpl will provide delicate lifetime control of filter chain context collections to achieve the goal of draining necessary connections on listener update.
Let me add how this concrete type is used

Reply:
I renamed.
WRT implement `Creator` in FilterChainManagerImpl, FilterChainManagerImpl need to refer a parent context. I don't know if it's making the logic clearer. I will update let you evaluate.

Reply:
removed